localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/parse.py

```diff
@@ -166,3 +169,15 @@ def fetch_stage_variables(context: RestApiInvocationContext) -> Optional[dict[st
             return None
 
         return stage_variables
+
+    @staticmethod
+    def populate_trace_id(headers: Headers) -> str:
+        incoming_trace = parse_trace_id(headers.get("x-amzn-trace-id", ""))
+        # parse_trace_id always return capitalized keys
+
+        trace = incoming_trace.get("Root", generate_trace_id())
+        incoming_parent = incoming_trace.get("Parent")
+        parent = incoming_parent or generate_trace_parent()
+        sampled = incoming_trace.get("Sampled", "1" if incoming_parent else "0")
+        # TODO: lineage? not sure what it related to
```

Comment:
Lineage is related to detecting recursive Lambda loops (July 2023): https://aws.amazon.com/blogs/compute/detecting-and-stopping-recursive-loops-in-aws-lambda-functions/

They extended support for S3, SQS, and SNS in Aug 2024: https://aws.amazon.com/blogs/compute/aws-lambda-introduces-recursive-loop-detection-apis/